File: arrayMaxConsecutiveSum/arrayMaxConsecutiveSum.ts
function arrayMaxConsecutiveSum(arr : number[], k : number) : number{
    let max = 0;
    let sum = 0;
        
    for (let i = 0; i < k; i++) {
        sum += arr[i];

    }
    max = sum;

    for(let i = k; i < arr.length; i++){
        sum -= arr[i - k];
        sum += arr[i];

        if(sum > max){
            max = sum;
        }
    }

    return max;
}

console.log(arrayMaxConsecutiveSum([2, 3, 5, 1, 6, 7, -1], 6));

File: arrayMaxConsecutiveSum/arrayMaxConsecutiveSum.java
package arrayMaxConsecutiveSum;

public class arrayMaxConsecutiveSum {
    
    public static int solution(int arr[], int k){
        int max = 0, sum = 0;
        
        for (int i = k; i < k; i++) {
            sum += arr[i];
        }

        max = sum;

        for(int i = k; i < arr.length; i++){
            sum -= arr[i - k];
            sum += arr[i];

            if(max < sum){
                max = sum;
            }
        }

        return max;
    }

    public static void main(String[] args) {
        int arr[] = {2, 3, 5, 1, 6};
        System.out.println(solution(arr, 2));
    }
}
